core/src/main/scala/org/apache/spark/scheduler/TaskScheduler.scala

```diff
@@ -54,9 +54,9 @@ private[spark] trait TaskScheduler {
   // Submit a sequence of tasks to run.
   def submitTasks(taskSet: TaskSet): Unit
 
-  // Kill all the tasks in a stage and fail the stage and all the jobs that depend on the stage.
+  // Kill all the tasks in all the stage attempts of the same stage Id
   // Throw UnsupportedOperationException if the backend doesn't support kill tasks.
-  def cancelTasks(stageId: Int, interruptThread: Boolean, reason: String): Unit
+  def killAllTaskAttempts(stageId: Int, interruptThread: Boolean, reason: String): Unit
```

Comment:
super nit: shall we put this method after `def killTaskAttempt`? The same for the testing `TaskScheduler` implementations to reduce the code diff.

Reply:
Updated, thanks!